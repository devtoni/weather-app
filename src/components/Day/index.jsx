import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

const Day = ({day, icon, info}) => {
  return (
    <Link to={{
      pathname: `/details/${day}`,
      state: {info}
    }}>
      <div>
        <img src={icon} alt='' />
        <h2>{day}</h2>
      </div>
    </Link>
  )
}

Day.propTypes = {
  day: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  info: PropTypes.object.isRequired
}

Day.defaultProps = {
  day: '',
  icon: '',
  info: {}
}

export default Day
