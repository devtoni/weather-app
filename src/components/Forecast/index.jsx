import React, { Component } from 'react'
import PropTypes from 'prop-types'
import weatherApi from '../../service/weatherApi'
import queryString from 'query-string'
import ListDays from '../ListDays'
import style from './main.scss'

class Forecast extends Component {
  constructor (props) {
    super(props)
    this.state = {
      days: ['']
    }
  }
  componentDidMount () {
    const { city } = queryString.parse(this.props.location.search)
    weatherApi.getForecast(city)
    .then(listDays => {
      this.setState({ days: listDays })
    })
  }
  render () {
    return (
      <div className={style.center}>
        <h2>{this.state.days[0].city}</h2>
        <ListDays days={this.state.days} />
      </div>
    )
  }
}

Forecast.propTypes = {
  location: PropTypes.shape({
    search: PropTypes.string.isRequired
  })
}

Forecast.defaultProps = {
  search: 'Barcelona'
}
export default Forecast
