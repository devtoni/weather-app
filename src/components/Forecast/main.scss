.center{
	margin-top: 40px;
	text-align:center;
}