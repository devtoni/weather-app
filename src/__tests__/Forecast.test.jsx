import React from 'react'
import { shallow } from 'enzyme'
import Api from '../service/weatherApi'
import toJson from 'enzyme-to-json'
import Forecast from '../components/Forecast/'
import Day from '../components/Day'

test('Forecast should render as expected', () => {
  const component = shallow(<Forecast />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Forecast has a prop search as expected by default', () => {
  const component = shallow(<Forecast />)
  console.log(component.instance().props) // Barcelona
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('when prop is passed by query, should render the expected result', () => {
  const component = shallow(<Forecast search='Barcelona' />)
  const {search} = component.instance().props
  const result = Api.getForecast(search)
  result.then((response) => {
    expect(component.find(Day).length).toEqual(response.data.list.length)
  })
})
