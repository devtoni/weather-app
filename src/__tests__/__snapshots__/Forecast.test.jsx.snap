// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast has a prop search as expected by default 1`] = `
<div
  className="center"
>
  <h2 />
  <ListDay
    days={
      Array [
        "",
      ]
    }
  />
</div>
`;

exports[`Forecast should render as expected 1`] = `
<div
  className="center"
>
  <h2 />
  <ListDay
    days={
      Array [
        "",
      ]
    }
  />
</div>
`;
